src/internal/deployment.js

```diff
@@ -39,11 +41,22 @@ class Deployment {
     this.githubServerUrl = context.githubServerUrl
     this.artifactName = context.artifactName
     this.isPreview = context.isPreview === true
+    this.timeout = MAX_TIMEOUT
+    this.startTime = null
   }
 
   // Ask the runtime for the unsigned artifact URL and deploy to GitHub Pages
   // by creating a deployment with that artifact
   async create(idToken) {
+    if (Number(core.getInput('timeout')) > MAX_TIMEOUT) {
+      core.warning(
+        `Warning: timeout value is greater than the allowed maximum - timeout set to the maximum of ${MAX_TIMEOUT} milliseconds.`
+      )
+    }
+
+    const TIMEOUT_INPUT = Number(core.getInput('timeout'))
+    this.timeout = !TIMEOUT_INPUT || TIMEOUT_INPUT <= 0 ? MAX_TIMEOUT : Math.min(TIMEOUT_INPUT, MAX_TIMEOUT)
```

Comment:
🧹 **Cleanup**

I see you took my other suggestion to heart. Nuanced style opinion would be that the uppercased-snakecase format is reserved for... not sure what the best word is... "true"/global/static constants? i.e. values that cannot be altered by the user, anyway. 😅 

Although `const` is the correct _keyword_ here, its usage more reflects that its value won't change during a single execution of this method, rather than never changing at all. Like I said, nuance. 🤷🏻 

```suggestion
    const timeoutInput = Number(core.getInput('timeout'))
    this.timeout = !timeoutInput || timeoutInput <= 0 ? MAX_TIMEOUT : Math.min(timeoutInput, MAX_TIMEOUT)
```

Reply:
We need a laughing emoji - that makes more sense! I'll switch it out again!